### ISSUE TITLE
Allocate new memory

### ISSUE THREAD

**Issue by JohnDeved** [Rob--/memoryjs#36]
First of all this library is amazing, ive been having tons of fun with it since Yesterday.

But while trying out writing bytearrays i noticed that it seems to be not possible to allocate new memory with this library. If you could allocate new memory in a process, then you could write way more Advanced assemly to inject into the game, since jumping to the new memory Location would give you tons of space to inject assemly.

Would it be possible to implement memory allocation in the library?

**Comment by Rob--** [comment]
Sure this is possible. If you want to write shell-code are you also looking for CreateRemoteThread functionality to execute the injected assembly?

**Comment by JohnDeved** [comment]
That would be cool!
I just wanted to make a godmode patch in the game i was testing this with and i noticed i was running out of space for the opcode.

**Comment by Rob--** [comment]
It's possible to expose any WinAPI functions through a Node Addon, when I get the time I'll expose VirtualAllocEx. Also working on DLL injection though! I'll post an update here when memory allocation has been implemented.

**Comment by JohnDeved** [comment]
Great! I'll be looking forward to it!

**Comment by Rob--** [comment]
@JohnDeved implemented in v3.2.3 (6e0496cf2e13713e43b502335bd726463b07300e and d778a60710401c6c745bb8795fa701e9e9f4c05b).

Not thoroughly tested, but seems to work according to the single test file I added. Use as follows:
```javascript
const processObject = memoryjs.openProcess('notepad.exe');

const address = memoryjs.virtualAllocEx(
  processObject.handle,
  null,
  0x60,
  memoryjs.MEM_RESERVE | memoryjs.MEM_COMMIT,
  memoryjs.PAGE_EXECUTE_READWRITE,
);

console.log(`Allocated address: 0x${address.toString(16).toUpperCase()}`);
```

Further documentation can be found in the [readme](https://github.com/Rob--/memoryjs#virtualallocexhandle-address-size-allocationtype-protection-callback). I highly recommend reading MSDN's [VirtualAllocEx](https://docs.microsoft.com/en-us/windows/desktop/api/memoryapi/nf-memoryapi-virtualprotectex) documentation.
